tests/unittests/tests-nanocoap/tests-nanocoap.c

```diff
@@ -1117,6 +1117,35 @@ static void test_nanocoap__token_length_ext_269(void)
     TEST_ASSERT_EQUAL_INT(14, hdr->ver_t_tkl & 0xf);
 }
 
+static void test_nanocoap__replace_etag(void)
```

Comment:
This function name does not explain your intend well enough. Remember to unit test _behavior_ so ideally something along `test_that_nanocoap_replaces_an_etag_within_a_prebuild_pkt_without_segfault` :smiling_imp: